File: csharp/isbn-verifier/IsbnVerifier.cs
using System;
using System.Text.RegularExpressions;

public static class IsbnVerifier
{
    public static bool IsValid(string number)
    {
        number = number.Replace("-", "");
        if(!IsLexicallyValid(number))
        {
            return false;
        }
        var sum = 0;
        var m = 10;
        int num;
        foreach (var x in number)
        {
            if(!Int32.TryParse(x.ToString(), out num))
            {
                num = 10;
            }
            sum += m * num;
            m--;
        }
        return sum % 11 == 0;
    }
    public static bool IsLexicallyValid(string num) => Regex.IsMatch(num, "^[0-9]{9}([0-9]|X)$");
}

File: javascript/isbn-verifier/isbn-verifier.js
export class ISBN {
  constructor(isbn) {
    this.isbn = isbn.replace(/-/g, "");
  }
  isValid() {
    return  this.isLexicallyValid() && this.isNumericallyValid();
  }
  isLexicallyValid(){
    return new RegExp(/^[0-9]{9}([0-9]|X)$/).test(this.isbn);
  }
  isNumericallyValid(){
    return [...this.isbn]
      .map((x, i) => x === "X" ? 10 * (this.isbn.length - i) : Number(x) * (this.isbn.length - i))
      .reduce((x, y) => x + y) % 11 === 0;
  }
}
